src/explorer.ts

```diff
@@ -213,3 +213,31 @@ class KubernetesNamespaceResource extends KubernetesResource {
         return treeItem;
     }
 }
+
+class KubernetesDeploymentFolder extends KubernetesResourceFolder {
+    constructor() {
+        super(kuberesources.allKinds.deployment);
+    }
+
+    async getChildren(kubectl: Kubectl, host: Host): Promise<KubernetesObject[]> {
+        const deployments = await kubectlUtils.getDeployments(kubectl);
+        return deployments.map((dp) => new KubernetesDeploymentResource(this.kind, dp.name, dp));
+    }
+}
+
+class KubernetesDeploymentResource extends KubernetesResource {
+    constructor(readonly kind: kuberesources.ResourceKind, readonly id: string, readonly metadata?: any) {
+        super(kind, id, metadata);
+    }
+
+    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        const treeItem = new vscode.TreeItem(this.id, vscode.TreeItemCollapsibleState.Collapsed);
+        treeItem.contextValue = `vsKubernetes.${this.id}`;
```

Comment:
Can we change the contextValue to `vsKubernetes.resource.${this.kind.abbreviation}`? The new value you provided will lose the registered context menus for deployment nodes.

Reply:
done.